src/configurator/question.rb

```diff
@@ -59,21 +59,12 @@ def use_readline?
         Metalware::Configurator.use_readline
       end
 
+      # Default for a boolean question which has a previous answer should be
+      # set to the input HighLine's `agree` expects, i.e. 'yes' or 'no'.
       def default_input
-        type.boolean? ? boolean_default_input : current_answer_value
-      end
-
-      def boolean_default_input
-        return nil if current_answer_value.nil?
-
-        # Default for a boolean question which has a previous answer should be
-        # set to the input HighLine's `agree` expects, i.e. 'yes' or 'no'.
-        current_answer_value ? 'yes' : 'no'
-      end
-
-      # The answer value this question at this level would currently take.
-      def current_answer_value
-        old_answer.nil? ? default : old_answer
+        return nil if default.nil?
+        return (default ? 'yes' : 'no') if type.boolean?
```

Comment:
I don't think inlining this is an improvement as it makes this method quite dense. Changing this to

```ruby
def default_input
  type.boolean? ? boolean_default_input : default
end
```
and adding back `boolean_default_input` as it was before would better separate what handling here is only needed for boolean questions (special handling for `nil` and using `yes` or `no`) from everything else (just using `default` as the default).

Reply:
Done